app/src/main/cpp/skyline/kernel/memory.h

```diff
@@ -255,14 +272,57 @@ namespace skyline {
             span<u8> CreateMirrors(const std::vector<span<u8>> &regions);
 
             /**
-             * @brief Frees the underlying physical memory for all full pages in the contained mapping
-             * @note All subsequent accesses to freed memory will return 0s
+             * @brief Sets the attributes for chunks within a certain range
+             */
+            void SetLockOnChunks(span<u8> memory, bool value);
+
+            void SetCPUCachingOnChunks(span<u8> memory, bool value);
+
+            /**
+             * @brief Sets the permission for chunks within a certain range
+             * @note The permissions set here are not accurate to the actual permissions set on the chunk and are only for the guest
+             */
+            void SetChunkPermission(span<u8> memory, memory::Permission permission);
+
+            /**
+             * @brief Gets the highest chunk's descriptor that contains this address
+             */
+            std::optional<std::pair<u8 *, ChunkDescriptor>> GetChunk(u8 *addr);
+
+            /**
+             * Various mapping functions for use by the guest
+             * @note UnmapMemory frees the underlying memory as well
```

Comment:
Add this comment to unmap memory rather than here.

Use a // comment for when you're trying to apply a comment to multiple functions